src/org/zalando/stups/friboo/system/new_http.clj

```diff
@@ -0,0 +1,275 @@
+    (merge this {:handler handler
+                 :httpd   (jetty/run-jetty handler (merge configuration {:join? false}))})))
+
+(defn stop-component
+  "Stops the Http component."
+  [{:as this :keys [handler httpd]}]
+  (if-not handler
+    (do
+      (log/debug "Skipping stop of Http because it's not running.")
+      this)
+    (do
+      (log/info "Stopping Http.")
+      (when httpd
+        (.stop httpd))
+      (dissoc this :handler :httpd))))
+
+(defrecord Http [;; parameters (filled in by make-http on creation)
+                  api-resource
+                  configuration
+                  security-handlers
+                  middlewares
+                  s1st-options
+                  ;; dependencies (filled in by the component library before starting)
+                  controller
+                  metrics
+                  audit-log
+                  ;; runtime vals (filled in by start-component)
+                  httpd
+                  handler]
+  Lifecycle
+
+  (start [this]
+    (start-component this))
+
+  (stop [this]
+    (stop-component this)))
+
+(defn redirect-to-swagger-ui
+  "Can be used as operationId for GET /"
+  [& _]
+  (ring.util.response/redirect "/ui/"))
+
+;; ## Middleware
+
+(defn wrap-default-content-type [next-handler content-type]
+  (fn [request]
+    (let [response (next-handler request)]
+      (if (get-in response [:headers "Content-Type"])
+        response
+        (assoc-in response [:headers "Content-Type"] content-type)))))
+
+(defn compute-request-info
+  "Creates a nice, readable request info text for logline prefixing."
+  [request]
+  (str
+    (.toUpperCase (-> request :request-method name))
+    " "
+    (:uri request)
+    " <- "
+    (if-let [x-forwarded-for (-> request :headers (get "x-forwarded-for"))]
+      x-forwarded-for
+      (:remote-addr request))
+    (if-let [tokeninfo (:tokeninfo request)]
+      (str " / " (get tokeninfo "uid") " @ " (get tokeninfo "realm"))
+      "")))
+
+(defn enrich-log-lines
+  "Adds HTTP request context information to the logging facility's MDC in the 'request' key."
+  [next-handler]
+  (fn [request]
+    (let [request-info (compute-request-info request)]
+      (with-logging-context
+        {:request (str " [" request-info "]")}
+        (next-handler request)))))
+
+(defn health-endpoint
+  "Adds a /.well-known/health endpoint for load balancer tests."
+  [handler]
+  (fn [request]
+    (if (= (:uri request) "/.well-known/health")
+      (-> (r/response "{\"health\": true}")
+          (ring/content-type-json)
+          (r/status 200))
+      (handler request))))
+
+(defn parse-basic-auth
+  "Parse HTTP Basic Authorization header"
+  [authorization]
+  (-> authorization
+      (clojure.string/replace-first "Basic " "")
+      .getBytes
+      b64/decode
+      String.
+      (clojure.string/split #":" 2)
+      (#(zipmap [:username :password] %))))
+
+;; TODO Zalando specific
```

Comment:
That and also I think it was added there because it was necessary for `pierone login`?